gamemodes/terrortown/gamemode/client/cl_player_ext.lua

```diff
@@ -412,19 +423,71 @@ hook.Add("CalcView", "TTT2ViewBobbingHook", function(ply, origin, angles, fov)
         return
     end
 
-    if (not ply:IsOnGround() and ply:WaterLevel() == 0) or ply:InVehicle() then
-        airtime = math.Clamp(airtime + 1, 0, 300)
-
-        return
-    end
-
     local view = {
         ply = ply,
         origin = origin,
         angles = angles,
         fov = fov,
     }
 
+    ply.lastFOVValue = ply.lastFOVValue or fov
+
+    local dynFOV = fov
+    local mul = ply:GetSpeedMultiplier() * SPRINT:HandleSpeedMultiplierCalculation(ply)
+    local desiredFOV = fov * mul ^ (1 / 6)
+
+    -- fixed mode: when SetZoom is set to something different than 0, this value should be used
+    -- without any custom modification
+    if ply:GetFOVFixedValue() then
+        dynFOV = fov
+
+    -- dynamic mode: transition between different FOV values
+    else
+        local time = math.max(
+            0,
+            (CurTime() - ply:GetFOVTime()) * game.GetTimeScale() * cvHostTimescale:GetFloat()
+        )
+
+        local progressTransition = math.min(1.0, time / ply:GetFOVTransitionTime())
+
+        -- if the transition progress has reached 100%, we should enable the sprint
+        -- FOV smoothing algorithm
+        if progressTransition >= 1.0 then
+            if desiredFOV > ply.lastFOVValue then
+                desiredFOV = math.min(desiredFOV, ply.lastFOVValue + FrameTime() * 40)
+            elseif desiredFOV < ply.lastFOVValue then
+                desiredFOV = math.max(desiredFOV, ply.lastFOVValue - FrameTime() * 40)
+            end
+
+            ply.lastFOVValue = desiredFOV
```

Comment:
Why does this not also use the last fov below from the network var? 

Regardless it is very confusing that we define another variable with the same name / purpose a few lines down.

Reply:
one of them is for the aimation on a frame by frame basis (`ply.lastFOVValue`) and the other is on a more marco level, that is used to handle the overall transition

Reply:
But do they differ in any way? Cant they be merged?

Reply:
no, they can't be merged. Let me explain:

the current transition progress is calculated by difference in the network vars for lastFOV and nextFOV times the time that passed. The result of that is applied to the current FOV. The result is also set to the players lastFOV value cache.
However this lastFOV value is not set on every frame. Sometimes it is intentionally kept on the last value

Reply:
Hmm then please try to rename the variables accordingly and maybe add a small comment explaining this. Otherwise its very confusing